### PR TITLE
Update and fix the dependencies in fpm

### PR DESCRIPTION
**Pull request by thfroitzheim**
Update and fix the fpm dependencies. Should be consistent among all projects for the xtb build.

### CODE REVIEW

File: config/meson.build [hunk 1]
```diff
@@ -142,7 +142,7 @@ endif
 # Create the tool chain library as subproject
 mctc_dep = dependency(
   'mctc-lib',
-  version: '>=0.3.0',
+  version: '=0.3.2',
```
Comment (marvinfriede) [inline config/meson.build:145]:
```suggestion
  version: '0.3.2',
```

File: config/meson.build [hunk 2]
```diff
@@ -151,6 +151,7 @@ lib_deps += mctc_dep
 # Create the electrostatic library as subproject
 multicharge_dep = dependency(
   'multicharge',
+  version: '=0.3.0',
```
Comment (marvinfriede) [inline config/meson.build:154]:
```suggestion
  version: '0.3.0',
```

File: test/unit/meson.build [hunk 3]
```diff
@@ -17,6 +17,7 @@
 # Create mstore as subproject for testing
 mstore_dep = dependency(
   'mstore',
+  version: '=0.3.0',
```
Comment (marvinfriede) [inline test/unit/meson.build:20]:
```suggestion
  version: '0.3.0',
```

File: config/meson.build [hunk 4]
```diff
@@ -142,7 +142,7 @@ endif
 # Create the tool chain library as subproject
 mctc_dep = dependency(
   'mctc-lib',
-  version: '>=0.3.0',
+  version: '0.3.2',
```
Comment (awvwgk) [inline config/meson.build:145]: This should have a lower bound of 0.3.0 unless we use an API introduced in a later version.

**Review by awvwgk — CHANGES_REQUESTED** [review]

File: config/meson.build (hunk 4 above)
Comment (thfroitzheim) [reply config/meson.build]: I don't think this is necessary here. Any mctc-lib version after 0.3.0 including head works. But we need at least multicharge v0.3.0 for the actinide tests.

**Review by awvwgk — APPROVED** [review]

**Review by marvinfriede — APPROVED** [review]

**Pull request merged**